internal/runtimehandlerhooks/high_performance_hooks.go

```diff
@@ -124,3 +183,71 @@ func setCPUSLoadBalancing(c *oci.Container, enable bool, schedDomainDir string)
 
 	return nil
 }
+
+func setIRQLoadBalancing(c *oci.Container, enable bool) error {
+	if c.Spec().Linux == nil ||
+		c.Spec().Linux.Resources == nil ||
+		c.Spec().Linux.Resources.CPU == nil ||
+		c.Spec().Linux.Resources.CPU.Cpus == "" {
+		return fmt.Errorf("failed to find the container %q CPUs", c.ID())
+	}
+
+	content, err := ioutil.ReadFile(irqSmpAffinityFile)
+	if err != nil {
+		return err
+	}
+	currentIRQSMPSetting := strings.Trim(string(content), "\n")
+
+	newIRQSMPSetting, err := UpdateIRQSmpAffinityMask(c.Spec().Linux.Resources.CPU.Cpus, currentIRQSMPSetting, enable)
+	if err != nil {
+		return err
+	}
+	newIRQSMPSetting += "\n"
+	ioutil.WriteFile(irqSmpAffinityFile, []byte(newIRQSMPSetting), 0644)
+	return nil
+}
+
+func setCPUQuota(ctx context.Context, parentDir string, c *oci.Container, enable bool) error {
+	original := c.Spec().Linux.CgroupsPath
+	var rpath string
+	if strings.Contains(original, ":") {
```

Comment:
can we consider to use some existing library to work with cgroups, I am pretty sure I saw some usage of such library under the repository `github.com/opencontainers/runc/libcontainer/cgroups`